ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/Constants.java

```diff
@@ -16,168 +16,79 @@
+  public static final int LATEST_INDEX_ROOTS_LENGTH = 8192; // 2^13 epochs ~36 days
+  public static final int LATEST_PENALIZED_EXIT_LENGTH = 8192; // 2^13 epochs ~36 days
 
   // Reward and penalty quotients
-  public static final int BASE_REWARD_QUOTIENT = (int) Math.pow(2, 10); // 1,024
-  public static final int WHISTLEBLOWER_REWARD_QUOTIENT = (int) Math.pow(2, 9); // 512
-  public static final int INCLUDER_REWARD_QUOTIENT = (int) Math.pow(2, 3); // 8
-  public static final int INACTIVITY_PENALTY_QUOTIENT = (int) Math.pow(2, 24); // 16,777,216
+  public static final int BASE_REWARD_QUOTIENT = 32; // 2^5
+  public static final int WHISTLEBLOWER_REWARD_QUOTIENT = 512; // 2^9
+  public static final int INCLUDER_REWARD_QUOTIENT = 8; // 2^3
+  public static final int INACTIVITY_PENALTY_QUOTIENT = 16777216; // 2^24
 
-  // Status codes
-  public static final int INITIATED_EXIT = 1;
-  public static final int WITHDRAWABLE = 2;
+  // Status flags
+  public static final int INITIATED_EXIT = 1; // 2^0
+  public static final int WITHDRAWABLE = 2; // 2^1
 
   // Max operations per block
-  public static final int MAX_PROPOSER_SLASHINGS = (int) Math.pow(2, 4); // 16
-  public static final int MAX_CASPER_SLASHINGS = (int) Math.pow(2, 4); // 16
-  public static final int MAX_ATTESTATIONS = (int) Math.pow(2, 7); // 128
-  public static final int MAX_DEPOSITS = (int) Math.pow(2, 4); // 16
-  public static final int MAX_EXITS = (int) Math.pow(2, 4); // 16
-
-  // Validator registry delta flags
-  public static final int ACTIVATION = 0;
-  public static final int EXIT = 1;
+  public static final int MAX_PROPOSER_SLASHINGS = 16; // 2^4
+  public static final int MAX_ATTESTER_SLASHINGS = 1; // 2^0
+  public static final int MAX_ATTESTATIONS = 128; // 2^7
+  public static final int MAX_DEPOSITS = 16; // 2^4
+  public static final int MAX_EXITS = 16; // 2^4
 
   // Signature domains
-  public static final UnsignedLong DOMAIN_DEPOSIT = UnsignedLong.valueOf(0);
-  public static final UnsignedLong DOMAIN_ATTESTATION = UnsignedLong.valueOf(1);
-  public static final UnsignedLong DOMAIN_PROPOSAL = UnsignedLong.valueOf(2);
-  public static final UnsignedLong DOMAIN_EXIT = UnsignedLong.valueOf(3);
-  public static final UnsignedLong DOMAIN_RANDAO = UnsignedLong.valueOf(4);
-  public static final int MIN_VALIDATOR_WITHDRAWAL_TIME =
-      (int) Math.pow(2, 14); // 16,384 slots ~27 hours
-  public static final int MAX_WITHDRAWALS_PER_EPOCH = 4;
-
-  public static final long GENESIS_SLOT = (long) Math.pow(2, 19);
-  public static final long GENESIS_EPOCH = GENESIS_SLOT / Constants.EPOCH_LENGTH;
+  public static final int DOMAIN_DEPOSIT = 0; //
+  public static final int DOMAIN_ATTESTATION = 1; //
+  public static final int DOMAIN_PROPOSAL = 2; //
+  public static final int DOMAIN_EXIT = 3; //
+  public static final int MAX_DOMAIN_RANDAOEXITS = 4; //
 
-  public static String getConstantsAsString() {
```

Comment:
Do we want this removed?

Reply:
We can add it back later. getConstantsAsString cannot be generated using the code generation tools.